src/main/java/at/hannibal2/skyhanni/features/mining/FlowstateHelper.kt

```diff
@@ -0,0 +1,123 @@
+package at.hannibal2.skyhanni.features.mining
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.event.HandleEvent
+import at.hannibal2.skyhanni.data.MiningAPI
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.mining.OreMinedEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.DelayedRun
+import at.hannibal2.skyhanni.utils.ItemUtils
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockItemModifierUtils.getEnchantments
+import at.hannibal2.skyhanni.utils.inPartialSeconds
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object FlowstateHelper {
+    private val config get() = SkyHanniMod.feature.mining
+    private var timeSinceBreak = SimpleTimeMark.farPast()
+    private var display: List<Renderable>? = null
+
+    private var blockBreakStreak = 0
+
+    private val pickobulusPattern by RepoPattern.pattern(
+        "mining.pickobulus.blockdestroy",
+        "§7Your §r§aPickobulus §r§7destroyed §r§e(?<amount>\\d+) §r§7blocks!"
+    )
+
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onBlockMined(event: OreMinedEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        timeSinceBreak = SimpleTimeMark.now().plus(10.seconds)
+        blockBreakStreak += event.extraBlocks.values.sum()
+        createDisplay()
+        attemptClearDisplay()
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!MiningAPI.inCustomMiningIsland()) return
+        if (hasFlowstate() == null) return
+
+        pickobulusPattern.matchMatcher(event.message) {
+            timeSinceBreak = SimpleTimeMark.now().plus(10.seconds)
+            blockBreakStreak += group("amount").toInt()
+            createDisplay()
+            attemptClearDisplay()
+        }
+    }
+
+    private fun attemptClearDisplay() {
+        DelayedRun.runDelayed(10.seconds) {
+            if (timeSinceBreak.isInFuture()) return@runDelayed
+            blockBreakStreak = 0
+            createDisplay()
+        }
+    }
```

Comment:
I have a bad feeling about this, Since you can break a lot of blocks in quick succession Which would launch a bunch of redundant delayed Runs. A simple onTick logic would be better in my opinion.

Reply:
can you check if the current implementation is more around what you had in mind

Reply:
Yeah it fine now. (with a minor logic error)